File: 27.07.2024/duplicates.py
def dupli(arr):
    seen = set()
    duplicates = set()
    for num in arr:
        if num in seen:
            duplicates.add(num)
        else:
            seen.add(num)
    return list(duplicates)

arr = [1, 2, 3, 4, 5, 1, 2]
print("Duplicates:",dupli(arr))

File: 27.07.2024/duplicates.c
#include <stdio.h>

void findDuplicates(int arr[], int n) {
    printf("Duplicates: ");
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] == arr[j]) {
                printf("%d ", arr[i]);
            }
        }
    }
    printf("\n");
}

int main() {
    int arr[] = {1, 2, 3, 4, 2, 3};
    int n = sizeof(arr) / sizeof(arr[0]);
    findDuplicates(arr, n);
    return 0;
}
